crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -187,19 +188,69 @@ where
         Ok(hash)
     }
 
-    async fn send_transaction(&self, request: TransactionRequest) -> EthResult<H256> {
+    async fn send_transaction(&self, mut request: TransactionRequest) -> EthResult<H256> {
         let from = match request.from {
             Some(from) => from,
             None => return Err(SignError::NoAccount.into()),
         };
+
+        // set nonce if not already set before
+        if request.nonce.is_none() {
+            let nonce =
+                self.get_transaction_count(from, Some(BlockId::Number(BlockNumberOrTag::Pending)))?;
+            request.nonce = Some(nonce.saturating_add(U256::from(1)))
```

Comment:
this should be checked_add and map error to `MaxNonce` see `EthApiError`

Reply:
yes